src/openforms/formio/components/custom.py

```diff
@@ -390,11 +391,53 @@ class AddressValueSerializer(serializers.Serializer):
         required=False,
         allow_blank=True,
     )
+    streetName = serializers.CharField(
+        label=_("street name"),
+        help_text=_("Found street name"),
+        required=False,
+        allow_blank=True,
+    )
+    city = serializers.CharField(
+        label=_("city"),
+        help_text=_("Found city"),
+        required=False,
+        allow_blank=True,
+    )
+    secretStreetCity = serializers.CharField(
+        label=_("city and street name secret"),
+        help_text=_("Secret for the combination of city and street name"),
+        required=False,
+        allow_blank=True,
+    )
 
     def validate_postcode(self, value: str) -> str:
         """Normalize the postcode so that it matches the regex from the BRK API."""
         return value.upper().replace(" ", "")
 
+    def validate(self, attrs):
+        attrs = super().validate(attrs)
+
+        city = attrs.get("city", "")
+        street_name = attrs.get("streetName", "")
+
+        if city and street_name:
+            existing_hmac = attrs.get("secretStreetCity", "")
+            postcode = attrs.get("postcode", "")
+            number = attrs.get("houseNumber", "")
+
+            computed_message = f"{postcode}/{number}/{city}/{street_name}"
+            computed_hmac = salted_hmac(
+                "location_check", value=computed_message
+            ).hexdigest()
+
+            if existing_hmac != computed_hmac:
```

Comment:
for best security practices (to protect against timing attacks), you want to use `django.utils.crypto.constant_time_compare` instead:

```suggestion
            if not constant_time_compare(existing_hmac, computed_hmac):
```